go/vt/schema/parser.go

```diff
@@ -122,7 +121,7 @@ func parseEnumOrSetTokens(enumOrSetValues string) (tokens []string) {
 		// input should not contain `enum(...)` column definition, just the comma delimited list
 		return tokens
 	}
-	tokens = textutil.SplitDelimitedList(enumOrSetValues)
+	tokens = strings.Split(enumOrSetValues, ",")
```

Comment:
Any reason(s) not to use the cvs encoding packing in the stdlib? https://pkg.go.dev/encoding/csv

Reply:
The main issue would be that quoting works differently there with `"` and not `'` like in SQL. Also escaping of quotes is different, so we'd have to write something specific for this or leverage instead `sqlparser` as @shlomi-noach mentioned.

Reply:
Yeah, I did notice that the csv encoder simply *removes* any commas in the values which surprised me. I do think that we should take this opportunity to address this more fully though as MySQL happily accepts enum string values with commas:
```
mysql> create table t1 (id int not null auto_increment primary key, ev enum('hi there', 'with, comma'));
Query OK, 0 rows affected (0.02 sec)

mysql> show create table t1;
+-------+-----------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------+
| Table | Create Table                                                                                                                                                                                        |
+-------+-----------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------+
| t1    | CREATE TABLE `t1` (
  `id` int NOT NULL AUTO_INCREMENT,
  `ev` enum('hi there','with, comma') DEFAULT NULL,
  PRIMARY KEY (`id`)
) ENGINE=InnoDB DEFAULT CHARSET=utf8mb4 COLLATE=utf8mb4_0900_ai_ci |
+-------+-----------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------+
1 row in set (0.01 sec)
```

Reply:
Right, there's a bunch of things that still fail here. I agree we should fix this properly.

Reply:
@shlomi-noach I pushed up a little specific parser for this in https://github.com/vitessio/vitess/pull/15493/commits/f90e797b0294b31d891209787197d91804ec78e8 that deals also with quotes inside the values as well.

I also added some low level test there with stuff like newlines, I think we should also add those to the end-to-end test too to try all the exotic things possible here.

Reply:
Realized my parser was not enough and also added a proper decoder for string values so we handle things line newlines correctly.